### ISSUE TITLE
az acr build results in "TypeError: 'NoneType' object is not callable"

### ISSUE THREAD

**Issue by jsmwoolf** [Azure/azure-cli#18790]
## Describe the bug

We use Azure container repository to store our docker images for Azure Web App.  At the end of our Jenkins build, we deploy a docker image over to one of our repositories.  However, we recently cannot upload Docker images due to an error in the CLI.

**Command Name**
`az acr build`

**Errors:**
```
The command failed with an unexpected error. Here is the traceback:
'NoneType' object is not callable
Traceback (most recent call last):
  File "/opt/az/lib/python3.6/site-packages/knack/cli.py", line 231, in invoke
    cmd_result = self.invocation.execute(args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 657, in execute
    raise ex
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 720, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 691, in _run_job
    result = cmd_copy(params)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 328, in __call__
    return self.handler(*args, **kwargs)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/command_operation.py", line 121, in handler
    return op(**command_args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/acr/build.py", line 109, in acr_build
    variant=platform_variant
TypeError: 'NoneType' object is not callable

```

## To Reproduce:

I'm only able to reproduce this error within a Jenkins slave instance.  We do login into azure via a service principle.  However, when logging into az with the same service principle, I'm able to upload a docker image on my local machine.  So, I'm not really sure how to reproduce outside of my Jenkins environment.

Here's a few steps that our Jenkins environment does during builds:

- Launch a Jenkins build and configure it to use Azure Virtual Machines
    - Slave machine should be running Ubuntu 18.04-LTS
    - The initialization script should contain the following line of code:  `curl -sL https://aka.ms/InstallAzureCLIDeb | sudo bash`
- Pipeline should contain lines of code that does the following Azure CLI commands (you can execute other Azure CLI commands)
    - `az login --service-principal -u {} -p {}--tenant {}`
    - `az acr build --registry {} --image {} --image {} --file Dockerfile-dev --build-arg {} --build-arg {} --build-arg {} .`

## Expected Behavior

Should be able to successfully upload docker images over to 

## Environment Summary
```
Linux-5.4.0-1051-azure-x86_64-with-debian-buster-sid
Python 3.6.10
Installer: DEB

azure-cli 2.26.0

```

For the Jenkin plugins installed:
```

Pipeline: Job (workflow-job): 2.41
Credentials Plugin (credentials): 2.5
JAXB plugin (jaxb): 2.3.0
Pipeline: Milestone Step (pipeline-milestone-step): 1.3.2
Pipeline: REST API Plugin (pipeline-rest-api): 2.19
JQuery3 API Plugin (jquery3-api): 3.6.0-1
GitHub Branch Source Plugin (github-branch-source): 2.11.1
Pipeline: Input Step (pipeline-input-step): 2.12
Matrix Authorization Strategy Plugin (matrix-auth): 2.6.7
Display URL API (display-url-api): 2.3.5
Pipeline: Step API (workflow-step-api): 2.23
Email Extension Plugin (email-ext): 2.83
CloudBees Credentials Plugin (cloudbees-credentials): 3.3
Pipeline: Multibranch (workflow-multibranch): 2.26
Pipeline: Stage Step (pipeline-stage-step): 2.5
Branch API Plugin (branch-api): 2.6.4
JavaScript GUI Lib: ACE Editor bundle plugin (ace-editor): 1.1
Popper.js 2 API Plugin (popper2-api): 2.5.4-2
ECharts API Plugin (echarts-api): 5.1.2-2
Pipeline: Stage Tags Metadata (pipeline-stage-tags-metadata): 1.8.5
Azure SDK API Plugin (azure-sdk): 23.v5682688d0eef
SCM API Plugin (scm-api): 2.6.4
Checks API plugin (checks-api): 1.7.0
Pipeline: Declarative (pipeline-model-definition): 1.8.5
Pipeline: Supporting APIs (workflow-support): 3.8
Copy Artifact Plugin (copyartifact): 1.46.1
Docker Commons Plugin (docker-commons): 1.17
REST API for Blue Ocean (blueocean-rest): 1.24.7
GIT server Plugin (git-server): 1.9
Azure VM Agents (azure-vm-agents): 783.v58077630847d
Bootstrap 5 API Plugin (bootstrap5-api): 5.0.1-2
Pipeline: SCM Step (workflow-scm-step): 2.13
Pipeline: Groovy (workflow-cps): 2.92
Build Blocker Plugin (build-blocker-plugin): 1.7.7
OWASP Markup Formatter Plugin (antisamy-markup-formatter): 2.1
JSch dependency plugin (jsch): 0.1.55.2
Azure Key Vault Plugin (azure-keyvault): 123.v1aba71c2d365
Mailer Plugin (mailer): 1.34
Bootstrap 4 API Plugin (bootstrap4-api): 4.6.0-3
Durable Task Plugin (durable-task): 1.37
Folders Plugin (cloudbees-folder): 6.15
Pipeline (workflow-aggregator): 2.6
GitHub API Plugin (github-api): 1.123
Resource Disposer Plugin (resource-disposer): 0.16
Jackson 2 API Plugin (jackson2-api): 2.12.3
Build Timestamp Plugin (build-timestamp): 1.0.3
Credentials Binding Plugin (credentials-binding): 1.26
Lockable Resources plugin (lockable-resources): 2.11
Extended Read Permission Plugin (extended-read-permission): 3.2
Cloud Statistics Plugin (cloud-stats): 0.27
Popper.js API Plugin (popper-api): 1.16.1-2
Git plugin (git): 4.7.2
Azure Container Agents Plugin (azure-container-agents): 209.vb415245ae23a
Java JSON Web Token (JJWT) Plugin (jjwt-api): 0.11.2-9.c8b45b8bb173
Azure Credentials (azure-credentials): 182.v3ccd4a755864
Structs Plugin (structs): 1.23
Azure Commons Plugin (azure-commons): 1.1.3
Plugin Utilities API Plugin (plugin-util-api): 2.3.0
Pipeline: Model API (pipeline-model-api): 1.8.5
Azure Container Registry Tasks Plugin (azure-container-registry-tasks): 0.6.5
Ant Plugin (ant): 1.11
Gradle Plugin (gradle): 1.36
PAM Authentication plugin (pam-auth): 1.6
Pipeline: Basic Steps (workflow-basic-steps): 2.23
Trilead API Plugin (trilead-api): 1.0.13
Token Macro Plugin (token-macro): 2.15
Snakeyaml API Plugin (snakeyaml-api): 1.29.1
JavaScript GUI Lib: Handlebars bundle plugin (handlebars): 3.0.8
SSH server (sshd): 3.0.3
Script Security Plugin (script-security): 1.77
Azure CLI Plugin (azure-cli): 0.9
GitHub plugin (github): 1.33.1
Oracle Java SE Development Kit Installer Plugin (jdk-tool): 1.5
Pipeline Graph Analysis Plugin (pipeline-graph-analysis): 1.11
Authentication Tokens API Plugin (authentication-tokens): 1.4
Pipeline: Build Step (pipeline-build-step): 2.13
LDAP Plugin (ldap): 2.7
Pipeline: API (workflow-api): 2.46
Docker API Plugin (docker-java-api): 3.1.5.2
Azure Storage plugin (windows-azure-storage): 358.v5c001416d74f
Command Agent Launcher Plugin (command-launcher): 1.6
Common API for Blue Ocean (blueocean-commons): 1.24.7
Apache HttpComponents Client 4.x API Plugin (apache-httpcomponents-client-4-api): 4.5.13-1.0
Pipeline: Shared Groovy Libraries (workflow-cps-global-lib): 2.21
Git client plugin (git-client): 3.7.2
Azure Container Service Plugin (azure-acs): 1.0.4
Font Awesome API Plugin (font-awesome-api): 5.15.3-3
Build Timeout (build-timeout): 1.20
Pipeline: GitHub Groovy Libraries (pipeline-github-lib): 1.0
Pipeline: Declarative Extension Points API (pipeline-model-extensions): 1.8.5
Pipeline: Nodes and Processes (workflow-durable-task-step): 2.39
Azure AD Plugin (azure-ad): 175.v5513346d764a
bouncycastle API Plugin (bouncycastle-api): 2.20
JUnit Plugin (junit): 1.51
Timestamper (timestamper): 1.13
WMI Windows Agents Plugin (windows-slaves): 1.8
Text Finder (text-finder): 1.16
Matrix Project Plugin (matrix-project): 1.19
Azure App Service Plugin (azure-app-service): 1.0.2
Caffeine API Plugin (caffeine-api): 2.9.1-23.v51c4e2c879c8
JavaScript GUI Lib: Moment.js bundle plugin (momentjs): 1.1.1
SSH Credentials Plugin (ssh-credentials): 1.19
OkHttp Plugin (okhttp-api): 3.14.9
Kubernetes Continuous Deploy Plugin (kubernetes-cd): 2.3.1
SSH Build Agents plugin (ssh-slaves): 1.32.0
Pipeline: Stage View Plugin (pipeline-stage-view): 2.19
Workspace Cleanup Plugin (ws-cleanup): 0.39
Plain Credentials Plugin (plain-credentials): 1.7
Result: [Plugin:workflow-job, Plugin:credentials, Plugin:jaxb, Plugin:pipeline-milestone-step, Plugin:pipeline-rest-api, Plugin:jquery3-api, Plugin:github-branch-source, Plugin:pipeline-input-step, Plugin:matrix-auth, Plugin:display-url-api, Plugin:workflow-step-api, Plugin:email-ext, Plugin:cloudbees-credentials, Plugin:workflow-multibranch, Plugin:pipeline-stage-step, Plugin:branch-api, Plugin:ace-editor, Plugin:popper2-api, Plugin:echarts-api, Plugin:pipeline-stage-tags-metadata, Plugin:azure-sdk, Plugin:scm-api, Plugin:checks-api, Plugin:pipeline-model-definition, Plugin:workflow-support, Plugin:copyartifact, Plugin:docker-commons, Plugin:blueocean-rest, Plugin:git-server, Plugin:azure-vm-agents, Plugin:bootstrap5-api, Plugin:workflow-scm-step, Plugin:workflow-cps, Plugin:build-blocker-plugin, Plugin:antisamy-markup-formatter, Plugin:jsch, Plugin:azure-keyvault, Plugin:mailer, Plugin:bootstrap4-api, Plugin:durable-task, Plugin:cloudbees-folder, Plugin:workflow-aggregator, Plugin:github-api, Plugin:resource-disposer, Plugin:jackson2-api, Plugin:build-timestamp, Plugin:credentials-binding, Plugin:lockable-resources, Plugin:extended-read-permission, Plugin:cloud-stats, Plugin:popper-api, Plugin:git, Plugin:azure-container-agents, Plugin:jjwt-api, Plugin:azure-credentials, Plugin:structs, Plugin:azure-commons, Plugin:plugin-util-api, Plugin:pipeline-model-api, Plugin:azure-container-registry-tasks, Plugin:ant, Plugin:gradle, Plugin:pam-auth, Plugin:workflow-basic-steps, Plugin:trilead-api, Plugin:token-macro, Plugin:snakeyaml-api, Plugin:handlebars, Plugin:sshd, Plugin:script-security, Plugin:azure-cli, Plugin:github, Plugin:jdk-tool, Plugin:pipeline-graph-analysis, Plugin:authentication-tokens, Plugin:pipeline-build-step, Plugin:ldap, Plugin:workflow-api, Plugin:docker-java-api, Plugin:windows-azure-storage, Plugin:command-launcher, Plugin:blueocean-commons, Plugin:apache-httpcomponents-client-4-api, Plugin:workflow-cps-global-lib, Plugin:git-client, Plugin:azure-acs, Plugin:font-awesome-api, Plugin:build-timeout, Plugin:pipeline-github-lib, Plugin:pipeline-model-extensions, Plugin:workflow-durable-task-step, Plugin:azure-ad, Plugin:bouncycastle-api, Plugin:junit, Plugin:timestamper, Plugin:windows-slaves, Plugin:text-finder, Plugin:matrix-project, Plugin:azure-app-service, Plugin:caffeine-api, Plugin:momentjs, Plugin:ssh-credentials, Plugin:okhttp-api, Plugin:kubernetes-cd, Plugin:ssh-slaves, Plugin:pipeline-stage-view, Plugin:ws-cleanup, Plugin:plain-credentials]
```

**Comment by yonzhan** [comment]
route to service team

**Comment by mmarabesi** [comment]
We are facing the same issue here with the altest version as well. Downgraded to version 2.23.0 and it is working now

**Comment by RyanS-C** [comment]
I have had this same issue today. My work around was to revert to CLI version 2.23

**Comment by codycodes** [comment]
If you're on Windows, you can most likely just install the earlier [release](https://github.com/Azure/azure-cli/releases)

For those like me who are using WSL or other *nix-based OS and are using `apt` as your package manager, here are the steps I took to downgrade:

```
# Use apt-cache to install different versions of a package

sudo apt-get update

# get the available versions using: apt-cache showpkg <package-name>
# install a specific version using: sudo apt-get install package=version -V

sudo apt-get install azure-cli=2.23.0-1~focal -V
```

More info: [How to Install Specific Version of Package using apt-get (linoxide.com)](https://linoxide.com/install-specific-version-package-apt-get/)

**Comment by northtyphoon** [comment]
The issue has been fixed since 2.26.1
